modules/bibdk_autocomplete/bibdk_autocomplete.module

```diff
@@ -39,98 +40,79 @@ function bibdk_autocomplete_menu() {
  * @param      $string
  * @param bool $retry -- the server tends to respond with a error 500 however
  *                       retrying once seems to fix it most of the times.
+ * @see: https://github.com/DBCDK/ortograf
  */
-function bibdk_autocomplete($p_uuid, $v_uuid, $string, $retry = false) {
+function bibdk_autocomplete($p_uuid, $v_uuid, $input, $retry = false) {
+  if (strlen($input) < 3) {
+    drupal_json_output(array());
+    drupal_exit();
+  }
   $items = array();
   $settings = variable_get('bibdk_autocomplete_settings', array());
   $url = variable_get('bibdk_autocomplete_rest_url', null);
   $vars = variable_get('bibdk_autocomplete_var', array());
   $value_vars = $vars[$p_uuid]['vars'][$v_uuid];
   unset($value_vars['autocomplete']);
-  if (empty($value_vars['index'])) {
-    watchdog('bibdk_autocomplete', 'Autocomplete index not defined for value uuid ' . $v_uuid, array(), $severity = WATCHDOG_WARNING);
-    drupal_json_output(array());
-    drupal_exit();
-  }
-
-  $filter_array = bibdk_autocomplete_parse_filters($string);
-  $search_string = key($filter_array);
-  $filter_string = bibdk_autocomplete_set_filter_query($filter_array[$search_string]);
-  if (!empty($filter_string)) {
-    if (!empty($value_vars['filterQuery'])) {
-      $value_vars['filterQuery'] .= ' AND ' . $filter_string;
-    } else {
-      $value_vars['filterQuery'] = $filter_string;
-    }
-  }
 
-  $settings['highlight'] = ($settings['highlight']) ? 'true' : null;
-  $options['query'] = array('query' => check_plain($search_string)) + $value_vars + $settings;
-  $options['maxTime'] = 300;
-
-  $url = ($value_vars['facetIndex']) ? $url . '/facetSpell' : $url . '/terms';
+  $options['query'] = array('q' => check_plain($input)) + $settings;
+  $options['query']['type'] = $value_vars['type'];
   $url = url($url, $options);
 
   watchdog('bibdk_autocomplete', 'Autocomplete URL sent: ' . $url, array(), $severity = WATCHDOG_DEBUG);
 
   $curl = new MicroCURL();
   $headers = array('Accept: application/json');
   $curl->set_option(CURLOPT_HTTPHEADER, $headers);
+  $curl_get = $curl->get(array($url));
   $result = json_decode($curl->get(array($url)));
```

Comment:
Hvorfor kalder du to gange på $url via curl? Du burde kalde den én gang og så decode resultatet.